plugins/node/opentelemetry-instrumentation-koa/src/utils.ts

```diff
@@ -36,6 +37,7 @@ export const getMiddlewareMetadata = (
         [SEMATTRS_HTTP_ROUTE]: layerPath?.toString(),
       },
       name: context._matchedRouteName || `router - ${layerPath}`,
+      layerName: context._matchedRouteName || layerPath?.toString() || '',
```

Comment:
Is there a strong need to have a separate `layerName` from the `name` here?  The `name` is the user-visible value used for the span name (IIUC), so it would be a little confusing to not match against that.

See also the matching handling that instrumentation-express does: supporting a string (exact match), regexp, or function for matching:
https://github.com/open-telemetry/opentelemetry-js-contrib/blob/80d0c74f8cafdfce681ef39e6cea103dca4e4ccd/plugins/node/opentelemetry-instrumentation-express/src/utils.ts#L120-L133

Reply:
That was because this logic ran after the name is mutated. This json is what I have when processing the metadata
``` json
{
    "attributes": {
        "koa.name": "/documents",
        "koa.type": "router",
        "http.route": "/documents"
    },
    "name": "router - /documents",
    "layerName": "/documents" <--- original layer name
}
```

so I used the `layerName`. I will change it to use the interpolated name instead.